vault/src/system.rs

```diff
@@ -478,7 +479,30 @@ impl VaultService {
         }
     }
 
+    async fn ensure_bitcoin_networks_match(&self) -> Result<(), Error> {
+        let bitcoin_network = self.btc_rpc_master_wallet.network().to_string();
+        let default_spec_bitcoin_network = &Value::default();
+        let system_properties = self.btc_parachain.get_rpc_properties().await.unwrap_or_default();
+        let parachain_bitcoin_network = system_properties
+            .get("bitcoinNetwork")
+            .unwrap_or(default_spec_bitcoin_network)
+            .as_str()
+            .unwrap_or_default()
+            .to_string();
+
+        // `parachain_bitcoin_network` can be `bitcoin-mainnet`, `bitcoin-testnet`, or `bitcoin-regtest`
+        // source: https://github.com/interlay/interbtc/blob/a71b970616b0a4a59cd2e709a606a9a78fce80ff/primitives/src/lib.rs#L23
+        // `bitcoin_network` can be `mainnet`, `testnet`, regtest.
+        // source: https://developer.bitcoin.org/reference/rpc/getblockchaininfo.html
+        if !parachain_bitcoin_network.contains(&bitcoin_network) {
```

Comment:
This change can be made backwards compatible if we don't error when the property does not exist and actually I would prefer to have that if we have old chain specs which don't specify that property.

Reply:
Done